File: C++/프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.cpp
#include <string>
#include <vector>
#include <regex>

using namespace std;

vector<int> solution(string s) {
    vector<int> answer(2);
    
    int oriLen = s.size();
    while(s != "1") {
        answer[0]++;
        s = regex_replace(s, regex("0"), "");
        answer[1] += (oriLen - s.size());
        s = to_string(stoull(bitset<32>(s.size()).to_string()));
        oriLen = s.size();
    }
    
    return answer;
}

File: Java/프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
class Solution {
    public int[] solution(String s) {
        int[] answer = new int[2];
        int oriLen = s.length();
        
        while(!s.equals("1")) {
            s = s.replaceAll("0", "");
            answer[0]++;
            answer[1] += (oriLen - s.length());
            s = Integer.toBinaryString(s.length());
            oriLen = s.length();
        }
        return answer;
    }
}